QueryBuilder/Compilers/Compiler.cs

```diff
@@ -45,16 +48,40 @@ protected Compiler()
 
         };
 
-        protected Dictionary<string, object> generateNamedBindings(object[] bindings)
+        protected IEnumerable<KeyValuePair<string, object>> generateNamedBindings(SqlResult ctx)
         {
-            return Helper.Flatten(bindings).Select((v, i) => new { i, v })
-                .ToDictionary(x => parameterPrefix + x.i, x => x.v);
+            int posCount = 0;
+            KeyValuePair<string, object> generateNamedBinding(object v)
+            {
+                object value = v;
+                if (v is Variable variable)
+                {
+                    try
+                    {
+                        value = ctx.Query.FindVariable(variable.Name);
+                    }
+                    catch (Exception) when (AllowUndefinedNamedVariables)
```

Comment:
I don't particularly like catching general exceptions, but I thought this might be useful if someone is just trying to generate a statement.